src/hexkit/providers/akafka/provider/eventsub.py

```diff
@@ -266,3 +464,231 @@ async def run(self, forever: bool = True) -> None:
         else:
             event = await self._consumer.__anext__()
             await self._consume_event(event)
+
+
+# Define function signature for the DLQ processor.
+DLQEventProcessor = Callable[[ConsumerEvent], Awaitable[Optional[ExtractedEventInfo]]]
+
+
+class DLQValidationError(RuntimeError):
+    """Raised when an event from the DLQ fails validation."""
+
+    def __init__(self, *, event: ConsumerEvent, error: str):
+        msg = f"DLQ Event '{get_event_label(event)}' is invalid: {error}"
+        super().__init__(msg)
+
+
+class DLQProcessingError(RuntimeError):
+    """Raised when an error occurs while processing an event from the DLQ."""
+
+    def __init__(self, *, event: ConsumerEvent, error: str):
+        msg = f"DLQ Event '{get_event_label(event)}' cannot be processed: {error}"
+        super().__init__(msg)
+
+
+def validate_dlq_headers(event: ConsumerEvent) -> None:
+    """Validate the headers that should be populated on every DLQ event.
+
+    Raises:
+    - `DLQValidationError`: If any headers are determined to be invalid.
+    """
+    headers = headers_as_dict(event)
+    expected_headers = ["type", "correlation_id", ORIGINAL_TOPIC_FIELD]
+    invalid_headers = [key for key in expected_headers if not headers.get(key, None)]
```

Comment:
```suggestion
    invalid_headers = [key for key in expected_headers if not headers.get(key)]
```

None is already the "default default".

Reply:
A better name would be `missing_headers` instead of `invalid_headers`.